GUI/src/app/encryption-components/menezes-vanstone/mv-client-panel/mv-client-panel.component.spec.ts

```diff
@@ -1,23 +1,255 @@
+                {provide: MvBackendRequestService, useValue: backendSpy},
+                {provide: DialogService, useValue: dialogSpy}
+            ]
+        })
+            .compileComponents();
+        backendRequestServiceSpy = TestBed.inject(MvBackendRequestService) as jasmine.SpyObj<MvBackendRequestService>;
+        dialogServiceSpy = TestBed.inject(DialogService) as jasmine.SpyObj<DialogService>;
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(MvClientPanelComponent);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should encrypt message when send is called', fakeAsync(() => {
+        const source_client = {
+            name: 'source_client',
+            keyPair: {
+                public_key: {
+                    curve: {
+                        a: 0,
+                        prime: '13',
+                        order_of_subgroup: '7',
+                        generator: {
+                            x: '3',
+                            y: '8',
+                            is_infinite: false
+                        }
+                    },
+                    y: {
+                        x: '3',
+                        y: '8',
+                        is_infinite: false
+                    }
+                },
+                private_key: {
+                    curve: {
+                        a: 0,
+                        prime: '19',
+                        order_of_subgroup: '13',
+                        generator: {
+                            x: '2',
+                            y: '7',
+                            is_infinite: false
+                        }
+                    },
+                    x: '3'
+                }
+            },
+            ciphertext: {
+                encrypted_message: 'encrypted_text',
+                points: [{x: '3', y: '8', is_infinite: false}]
+            },
+            plaintext: 'Test Message 12 ! B',
+            signature: {
+                r: '123',
+                s: '456'
+            },
+            sendingTo: undefined,
+            receivedFrom: undefined,
+            signature_valid: 'ungeprüft'
+        };
+
+        const expectedCipherText: MvCipherText = {
+            encrypted_message: 'encrypted_text',
+            points: [{x: '3', y: '8', is_infinite: false}]
+        };
+        const expectedSignature: MvSignature = {r: '123', s: '456'};
+
+        backendRequestServiceSpy.encrypt.and.returnValue(of(expectedCipherText));
+        backendRequestServiceSpy.sign.and.returnValue(of(expectedSignature));
+        dialogServiceSpy.startTimedCalc.and.returnValue('loadingKey');
+
+        component.client = source_client;
+        component.client.sendingTo = source_client;
+        component.encrypt();
+        tick(); // Simuliert asynchrone Operationen
```

Comment:
```suggestion
        tick();
```
Nicht notwendig, ist klar, weil fakeAsync